contrib/repro_test.bzl

```diff
@@ -0,0 +1,243 @@
+    img_outs_repro = ctx.outputs.img_outs_repro
+
+    # Build and extract test_image inside a container.
+    _extract.implementation(
+        ctx,
+        name = "build_test_image_and_extract",
+        commands = commands,
+        extract_file = extract_path,
+        image = image_tar,
+        docker_run_flags = docker_run_flags,
+        output_file = img_outs,
+        script_file = ctx.actions.declare_file(name + ".build"),
+    )
+
+    # Build and extract test_image again in a different container to try and
+    # repro previously built test_image.
+    _extract.implementation(
+        ctx,
+        name = "build_test_image_and_extract_repro",
+        commands = commands,
+        extract_file = extract_path,
+        image = image_tar,
+        docker_run_flags = docker_run_flags,
+        output_file = img_outs_repro,
+        script_file = ctx.actions.declare_file(name + "_repro.build"),
+    )
+
+    # Expand template to run the image comparison test.
+    diff_tool_exec = ctx.executable._container_diff_tool
+    ctx.actions.expand_template(
+        template = ctx.file._test_tpl,
+        substitutions = {
+            "%{container_diff_args}": " ".join(ctx.attr.container_diff_args),
+            "%{container_diff_tool}": diff_tool_exec.short_path,
+            "%{img1_path}": img_outs.short_path,
+            "%{img2_path}": img_outs_repro.short_path,
+            "%{img_name}": test_img_label.name,
+        },
+        output = ctx.outputs.test_script,
+        is_executable = True,
+    )
+
+    runfiles = ctx.runfiles(files = [img_outs, img_outs_repro, diff_tool_exec])
+
+    return [
+        DefaultInfo(
+            executable = ctx.outputs.test_script,
+            runfiles = runfiles,
+        ),
+    ]
+
+container_repro_test = rule(
+    attrs = dicts.add(_container.image.attrs, {
+        "test_image": attr.label(
+            doc = "A container_image target to test for reproducibility. " +
+                  "Must be a label relative to the root of the project and " +
+                  "cannot be relative to the container_repro_test target.",
+            mandatory = True,
+        ),
+        "src_project_files": attr.label(
+            allow_files = True,
+            mandatory = True,
+            doc = "Files of the source project where the test_image target " +
+                  "lives. The WORKSPACE file must be specified and is " +
+                  "assumed to be at the root of the project.",
+        ),
+        "base": attr.label(
+            allow_files = container_filetype,
+            mandatory = True,
+            doc = "An image target compatible with the `base` attribute of " +
+                  "the container_image rule to build and reproduce the " +
+                  "test_image in.",
+        ),
+        "container_diff_args": attr.string_list(
+            default = [
+                "--type=file",
+                "--type=size",
+                "--type=apt",
```

Comment:
should we support anything other than overriding --type args? If not, lets only take as input the RHS of these ( ["file","size","apt","pip"]) and make the interface narrower.

Reply:
Done
Using just the --type flag now should be sufficient for now.